tools/gen_web_locale_keymap/lib/common.dart

```diff
@@ -159,22 +159,20 @@ class _StringStream {
 
 Map<String, int> _unmarshallCodeMap(_StringStream stream) {
   final int entryNum = stream.readIntAsVerbatim();
-  return Map<String, int>.fromEntries((() sync* {
-    for (int entryIndex = 0; entryIndex < entryNum; entryIndex += 1) {
-      yield MapEntry<String, int>(stream.readEventKey(), stream.readIntAsChar());
-    }
-  })());
+  return Map<String, int>.fromEntries(
+    Iterable<int>.generate(entryNum).map((final int index) =>
+      MapEntry<String, int>(stream.readEventKey(), stream.readIntAsChar())
```

Comment:
Here and below, can we pass the generator function as the second argument to `generate`? `generate` may be lazy, but why risk it? Unless there is some reason we can't use the generator function here.